### PR TITLE
feat(sucrase): better TS intellisense & export typings 

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
## Rollup Plugin Name: `sucrase`

This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [x] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description

TLDR; added typings for Sucrase & convert `src/index.js` to `src/index.ts`

### CODE REVIEW

File: packages/sucrase/package.json [hunk 1]
```diff
@@ -66,5 +67,12 @@
...
     ]
-  }
+  },
+  "contributors": [
+    {
+      "name": "Vinayak Kulkarni",
+      "email": "inbox.vinayak@gmail.com",
+      "url": "https://vinayakkulkarni.dev"
+    }
+  ]
```
Comment (shellscape) [inline packages/sucrase/package.json:77]: Please remove this block. I understand that you'd like credit for the change, but we don't add individual contributors to plugins' `package.json` files. You'll be recognized here: https://github.com/rollup/plugins/graphs/contributors and our `LICENSE` references that page specifically.

File: packages/sucrase/src/index.ts [hunk 2]
```diff
@@ -28,7 +30,7 @@ module.exports = function sucrase(opts = {}) {
       }
     },
 
-    transform(code, id) {
+    transform(code: any, id: any) {
```
Comment (shellscape) [inline packages/sucrase/src/index.ts:33]: if we're going to type this, let's not use `any`. If it's too heavy a lift to type properly, please revert.

File: packages/sucrase/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Plugin } from 'rollup';
+type Options = {
```
Comment (shellscape) [inline packages/sucrase/types/index.d.ts:2]:
```suggestion
type RollupSucraseOptions = {
```

to follow convention in the repo

**Review by shellscape — CHANGES_REQUESTED** [review]
Thanks for the PR. You'll also need to add the `types` directory to `files` in `package.json` or it won't ship with the types.

File: packages/sucrase/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  transforms?: string[];
+  include?: string[];
+  exclude?: string[];
```
Comment (shellscape) [inline packages/sucrase/types/index.d.ts:5]: these should use `FilterPattern` https://github.com/rollup/plugins/blob/ca0540f30ef3afd5bd998132e6b5931b0c319234/packages/pluginutils/types/index.d.ts#L23

**Comment by shellscape** [comment]
Actually, just realized that this is superseded by #898. Thanks for your work on this, but we'll be going with the other PR.

**Comment by vinayakkulkarni** [comment]
> Actually, just realized that this is superseded by #898. Thanks for your work on this, but we'll be going with the other PR.

Sweet! As long as we get the typings.. Alls well :)
